engine/rpcserver.go

```diff
@@ -3072,3 +3044,39 @@ func (s *RPCServer) GetRecentTrades(_ context.Context, r *gctrpc.GetSavedTradesR
 
 	return resp, nil
 }
+
+func (s *RPCServer) checkParams(exch string, a asset.Item, p currency.Pair) (exchange.IBotExchange, error) {
+	e := s.GetExchangeByName(exch)
+	if e == nil {
+		return nil, fmt.Errorf("exchange %s is not supported", exch)
+	}
+	if !e.IsEnabled() {
+		return nil, fmt.Errorf("exchange %s is not enabled", exch)
```

Comment:
Please make all of these errors actual errors and then wrap them. eg
`errExchangeNotEnabled = errors.New("exchange is not enabled")`
`fmt.Errorf("%w %s", errExchangeNotEnabled, exch)`

This allows you to do better error checks in tests or inspect errors at runtime